### ISSUE TITLE
inception-v3 result train-acc and val-acc

### ISSUE THREAD

**Issue by qiaohaijun** [dmlc/mxnet#1399]
16 machines with 4 k40m gpus
the config is:
batch_size=256, 
clip_gradient=5.0, 
data_dir='/search/data/',
data_shape=299, 
gpus='0,1,2,3', 
kv_store='dist_sync', 
log_dir='/tmp/', 
log_file='inception-v3-16m-try-1.log', 
lr=0.005, 
lr_factor=0.94, 
lr_factor_epoch=1.0, 
mean_file='2w_mean.bin', 
model_prefix='model/model-2w-384', 
network='inception-v3', 
num_classes=21841, 
num_epochs=100, 
num_examples=14192019, 
train_dataset='imagenet_2w_rec_resize_384/', 
val_dataset='imagenet_1k_val_resize_384.rec'

---

2016-01-29 04:28:53,210 Node[0] Epoch[0] Train-accuracy=0.028916
2016-01-29 08:10:33,458 Node[0] Epoch[1] Train-accuracy=0.110923
2016-01-29 11:52:04,246 Node[0] Epoch[2] Train-accuracy=0.163970
2016-01-29 15:33:39,505 Node[0] Epoch[3] Train-accuracy=0.198715
2016-01-29 19:15:12,641 Node[0] Epoch[4] Train-accuracy=0.223740
2016-01-29 22:56:34,957 Node[0] Epoch[5] Train-accuracy=0.242720
2016-01-30 02:37:49,585 Node[0] Epoch[6] Train-accuracy=0.258250
2016-01-30 06:19:21,172 Node[0] Epoch[7] Train-accuracy=0.270953
2016-01-30 10:01:06,645 Node[0] Epoch[8] Train-accuracy=0.281195
2016-01-30 13:42:49,914 Node[0] Epoch[9] Train-accuracy=0.291307
2016-01-30 17:23:37,076 Node[0] Epoch[10] Train-accuracy=0.299192
2016-01-30 21:05:13,990 Node[0] Epoch[11] Train-accuracy=0.307120
2016-01-31 00:46:21,347 Node[0] Epoch[12] Train-accuracy=0.313478
2016-01-31 04:27:37,291 Node[0] Epoch[13] Train-accuracy=0.319380
2016-01-31 08:09:12,578 Node[0] Epoch[14] Train-accuracy=0.325534
2016-01-31 11:50:41,690 Node[0] Epoch[15] Train-accuracy=0.329648
2016-01-31 15:31:41,191 Node[0] Epoch[16] Train-accuracy=0.334178
2016-01-31 19:13:12,318 Node[0] Epoch[17] Train-accuracy=0.338781
2016-01-31 22:55:10,268 Node[0] Epoch[18] Train-accuracy=0.342307
2016-02-01 02:36:26,516 Node[0] Epoch[19] Train-accuracy=0.345871
2016-02-01 06:17:50,365 Node[0] Epoch[20] Train-accuracy=0.350035
2016-02-01 09:59:10,152 Node[0] Epoch[21] Train-accuracy=0.352539

---

2016-01-29 04:29:06,522 Node[0] Epoch[0] Validation-accuracy=0.088867
2016-01-29 08:10:46,579 Node[0] Epoch[1] Validation-accuracy=0.172526
2016-01-29 11:52:17,664 Node[0] Epoch[2] Validation-accuracy=0.230143
2016-01-29 15:33:52,862 Node[0] Epoch[3] Validation-accuracy=0.261719
2016-01-29 19:15:25,818 Node[0] Epoch[4] Validation-accuracy=0.289388
2016-01-29 22:56:48,262 Node[0] Epoch[5] Validation-accuracy=0.304362
2016-01-30 02:38:03,724 Node[0] Epoch[6] Validation-accuracy=0.322115
2016-01-30 06:19:34,461 Node[0] Epoch[7] Validation-accuracy=0.335286
2016-01-30 10:01:19,961 Node[0] Epoch[8] Validation-accuracy=0.339193
2016-01-30 13:43:03,251 Node[0] Epoch[9] Validation-accuracy=0.348958
2016-01-30 17:23:50,598 Node[0] Epoch[10] Validation-accuracy=0.360352
2016-01-30 21:05:27,273 Node[0] Epoch[11] Validation-accuracy=0.362956
2016-01-31 00:46:34,635 Node[0] Epoch[12] Validation-accuracy=0.364258
2016-01-31 04:27:51,611 Node[0] Epoch[13] Validation-accuracy=0.369591
2016-01-31 08:09:25,763 Node[0] Epoch[14] Validation-accuracy=0.366211
2016-01-31 11:50:55,162 Node[0] Epoch[15] Validation-accuracy=0.381510
2016-01-31 15:31:54,725 Node[0] Epoch[16] Validation-accuracy=0.377279
2016-01-31 19:13:25,490 Node[0] Epoch[17] Validation-accuracy=0.385742
2016-01-31 22:55:23,616 Node[0] Epoch[18] Validation-accuracy=0.389323
2016-02-01 02:36:40,061 Node[0] Epoch[19] Validation-accuracy=0.394206
2016-02-01 06:18:04,726 Node[0] Epoch[20] Validation-accuracy=0.393630
2016-02-01 09:59:23,439 Node[0] Epoch[21] Validation-accuracy=0.399414

----
2016-01-29 04:28:53,210 Node[0] Epoch[0] Time cost=13259.672
2016-01-29 08:10:33,458 Node[0] Epoch[1] Time cost=13285.845
2016-01-29 11:52:04,246 Node[0] Epoch[2] Time cost=13276.579
2016-01-29 15:33:39,505 Node[0] Epoch[3] Time cost=13280.769
2016-01-29 19:15:12,641 Node[0] Epoch[4] Time cost=13278.707
2016-01-29 22:56:34,957 Node[0] Epoch[5] Time cost=13268.017
2016-01-30 02:37:49,586 Node[0] Epoch[6] Time cost=13260.201
2016-01-30 06:19:21,172 Node[0] Epoch[7] Time cost=13276.348
2016-01-30 10:01:06,645 Node[0] Epoch[8] Time cost=13291.060
2016-01-30 13:42:49,914 Node[0] Epoch[9] Time cost=13288.862
2016-01-30 17:23:37,076 Node[0] Epoch[10] Time cost=13232.742
2016-01-30 21:05:13,991 Node[0] Epoch[11] Time cost=13282.287
2016-01-31 00:46:21,347 Node[0] Epoch[12] Time cost=13252.956
2016-01-31 04:27:37,292 Node[0] Epoch[13] Time cost=13261.529
2016-01-31 08:09:12,578 Node[0] Epoch[14] Time cost=13279.819
2016-01-31 11:50:41,690 Node[0] Epoch[15] Time cost=13274.849
2016-01-31 15:31:41,191 Node[0] Epoch[16] Time cost=13244.960
2016-01-31 19:13:12,318 Node[0] Epoch[17] Time cost=13276.502
2016-01-31 22:55:10,268 Node[0] Epoch[18] Time cost=13303.707
2016-02-01 02:36:26,516 Node[0] Epoch[19] Time cost=13261.797
2016-02-01 06:17:50,366 Node[0] Epoch[20] Time cost=13269.179
2016-02-01 09:59:10,152 Node[0] Epoch[21] Time cost=13264.311

**Comment by qiaohaijun** [comment]
@mli

**Comment by piiswrong** [comment]
Why is Val acc higher then train acc when there are no drop outs

**Comment by qiaohaijun** [comment]
I use the 1k val dataset。

**Comment by antinucleon** [comment]
No seems you have done something wrong. Could you paste your data iter conf?

**Comment by qiaohaijun** [comment]
batch_size=256, 
clip_gradient=5.0, 
data_dir='/search/data/',
data_shape=299, 
gpus='0,1,2,3', 
kv_store='dist_sync', 
log_dir='/tmp/', 
log_file='inception-v3-16m-try-1.log', 
lr=0.005, 
lr_factor=0.94, 
lr_factor_epoch=1.0, 
mean_file='2w_mean.bin', 
model_prefix='model/model-2w-384', 
network='inception-v3', 
num_classes=21841, 
num_epochs=100, 
num_examples=14192019, 
train_dataset='imagenet_2w_rec_resize_384/', 
val_dataset='imagenet_1k_val_resize_384.rec'

**Comment by antinucleon** [comment]
OK. I made a mistake. If you train on ImageNet-22k and val on 1k, it looks not bad for your current result. https://mxnet.readthedocs.org/en/latest/tutorial/imagenet_full.html

**Comment by qiaohaijun** [comment]
@antinucleon thx your answer

**Comment by mli** [comment]
should the lr be so small? we often use lr=0.05. also we found

1. python's image resize function could affect the accuracy
2. use cudnn v4 instead of v3, the latter has bug on the pooling layer 

i think @antinucleon knows more about the issues.

**Comment by qiaohaijun** [comment]
@mli I will try cudnnv4

**Comment by yuzcccc** [comment]
totally 64 k40m...? orzzzz
